File: csharp/kindergarten-garden/KindergartenGarden.cs
using System;
using System.Collections.Generic;
using System.Linq;

public enum Plant
{
    Violets = 'V',
    Radishes = 'R',
    Clover = 'C',
    Grass = 'G'
}

public class KindergartenGarden
{
    private const int rowsPerStudent = 2;
    private readonly string diagram;
    private readonly string[] students = @"Alice, Bob, Charlie, David, Eve, Fred, Ginny, 
                                  Harriet, Ileana, Joseph, Kincaid, Larry".Split(", ");

    public KindergartenGarden(string diagram)
    {
        this.diagram = diagram;
    }

    public IEnumerable<Plant> Plants(string student)
    {
        var offset = Array.IndexOf(students, student) * rowsPerStudent;

        return diagram.Split("\n")
                .SelectMany(row => row.Substring(offset, rowsPerStudent))
                .Select(p => (Plant)p);
    }
}

File: fsharp/kindergarten-garden/KindergartenGarden.fs
module KindergartenGarden

type Plant = Grass | Clover | Radishes | Violets

let students = 
    [| "Alice"; "Bob"; "Charlie"; "David"; 
    "Eve"; "Fred"; "Ginny"; "Harriet"; 
    "Ileana"; "Joseph"; "Kincaid"; "Larry" |]

let charToPlant = function
    | 'G' -> Grass
    | 'C' -> Clover
    | 'R' -> Radishes
    | 'V' -> Violets
    | _ -> failwith "Some unknown abomination of a flower."

let plants (diagram: string) (student: string) =
    let ind = Array.findIndex ((=) student) students

    diagram.Split '\n'
    |> Seq.collect (Seq.skip (ind * 2) >> Seq.take 2)
    |> Seq.map charToPlant
    |> Seq.toList